kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -542,6 +659,216 @@ namespace Kratos
+			}
+		}
+
+		// Check if element qualifies as being intersected after calculating extrapolated intersections and calculate elemental distances
+		if (n_cut_edges >= TDim) {
+			// Calculate points from nodes of edges and length ratio of intersections
+			std::vector<array_1d <double,3> > intsect_pts_vector;
+			ConvertRatiosToIntersectionPoints(r_geometry, combined_edge_ratios, intsect_pts_vector);
+
+			// If there are more than 3 intersected edges (3D), compute the least squares plane approximation
+			// by using the ComputePlaneApproximation utility. Otherwise, the distance is computed using
+			// the plane defined by the 3 intersection points (3D).
+			const bool do_plane_approx = (n_cut_edges == TDim) ? false : true;
+
+			if (do_plane_approx){
+				// Call the plane optimization utility
+				array_1d<double,3> base_pt, normal;
+				ComputePlaneApproximation(rElement, intsect_pts_vector, base_pt, normal);
+
+				// Compute the distance to the approximation plane
+				Plane3D approximation_plane(normal, Point{base_pt});
+				for (uint8_t i = 0; i < n_nodes; i++) {
+					rElementalDistancesExtraVector[i] = approximation_plane.CalculateSignedDistance(r_geometry[i]);
+				}
+			} else {
+				// Create a plane with the 3 intersection points (or 2 in 2D)
+				Plane3D plane = SetIntersectionPlane(intsect_pts_vector);
+
+				// Compute the distance to the intersection plane
+				for (uint8_t i = 0; i < n_nodes; i++) {
+					rElementalDistancesExtraVector[i] = plane.CalculateSignedDistance(r_geometry[i]);
+				}
+			}
+
+			// Correct the distance values orientation
+			CorrectDistanceOrientation(r_geometry, rIntersectedObjects, rElementalDistancesExtraVector);
+		}
+
+	}
+
+	template<std::size_t TDim>
+	void CalculateDiscontinuousDistanceToSkinProcess<TDim>::ConvertRatiosToIntersectionPoints(
+        const Element::GeometryType& rGeometry,
+        const array_1d<double, (TDim == 2) ? 3 : 6> &rEdgeRatiosVector,
+        std::vector<array_1d <double,3> > &rIntersectionPointsVector)
+	{
+		constexpr std::size_t n_edges = (TDim == 2) ? 3 : 6;
+		const auto r_edges_container = rGeometry.GenerateEdges();
+
+		// Clear intersection points vector
+		rIntersectionPointsVector.clear();
+
+		// Calculate intersection point of each edge that is intersected
+		for (std::size_t i_edge = 0; i_edge < n_edges; ++i_edge){
+			if (rEdgeRatiosVector[i_edge] >= 0.0){
+				array_1d<double,3> avg_pt = ConvertEdgeRatioToIntersectionPoint(r_edges_container[i_edge], rEdgeRatiosVector[i_edge]);
+				rIntersectionPointsVector.push_back(avg_pt);
+			}
+		}
+	}
+
+	template<std::size_t TDim>
+	double CalculateDiscontinuousDistanceToSkinProcess<TDim>::ConvertIntersectionPointToEdgeRatio(
+        const Kratos::Geometry<Kratos::Node<3> >& rEdge,
+        const array_1d<double,3>& rIntersectionPoint)
+	{
+		const double edge_length = rEdge.Length();
+		const double dist_avg_pt = norm_2(rEdge[0] - rIntersectionPoint);
+		return dist_avg_pt / edge_length;
```

Comment:
Shouldn't we check in here that `edge_length` is not close to zero?

Reply:
Mmm, what would we do if it is close to zero? And what would be the limit?
I'm thinking if edge_length is close to zero, then dist_avg_pt even more so and we would still get our ratio.
You mean to check, that edge_length is large enough to still be represented and bigger than a numerical error?

Reply:
If the `edge_length` is close to zero we'll get a `Nan`. I'd throw an error if the value is close to the machine precision.

Reply:
So I check for the epsilon I saw at other places?

Reply:
I think this makes sense.